packages/beacon-node/src/chain/produceBlock/produceBlockBody.ts

```diff
@@ -616,4 +564,81 @@ function preparePayloadAttributes(
   return payloadAttributes;
 }
 
-/** process_sync_committee_contributions is implemented in syncCommitteeContribution.getSyncAggregate */
+export async function produceCommonBlockBody<T extends BlockType>(
+  this: BeaconChain,
+  blockType: T,
```

Comment:
Why do we need to pass blockType here? Shouldn't be relevant for this function

Reply:
So we can pass call this for either type of block.

Reply:
But why do we need to know the block type at all? The function produces the same result irrespective of the block type we pass here (or at least that's what I understood is the goal of this PR, to consolidate body production and only do it once).

Reply:
As of current implementation we need to know block type to update respective metric. Once we consolidate metrics then we don't need block type for common block body.

Reply:
Let's keep metrics as is for now then, will give this another review locally as it is probably easier that way